system/RESTful/ResourcePresenter.php

```diff
@@ -0,0 +1,204 @@
+ *
+ * @package CodeIgniter\RESTful
+ */
+class ResourcePresenter extends Controller
+{
+
+	/**
+	 *
+	 * @var string Name of the model class managing this resource's data
+	 */
+	protected $modelName = null;
+
+	/**
+	 *
+	 * @var \CodeIgniter\Model the model holding this resource's data
+	 */
+	protected $model = null;
+
+	//--------------------------------------------------------------------
+
+	public function initController(RequestInterface $request, ResponseInterface $response, LoggerInterface $logger)
+	{
+		parent::initController($request, $response, $logger);
+
+		// instantiate our model, if needed
+		$this->setModel($this->modelName);
+	}
+
+	//--------------------------------------------------------------------
+
+	/**
+	 * Present a view of resource objects
+	 *
+	 * @return string
+	 */
+	public function index()
+	{
+		return lang('RESTful.notImplemented', ['index']);
+	}
+
+	/**
+	 * Present a view to present a specific resource object
+	 *
+	 * @param  type $id
+	 * @return string
+	 */
+	public function show($id = null)
+	{
+		return lang('RESTful.notImplemented', ['show']);
+	}
+
+	/**
+	 * Present a view to present a new single resource object
+	 *
+	 * @param  type $id
+	 * @return string
+	 */
+	public function new()
+	{
+		return lang('RESTful.notImplemented', ['new']);
+	}
+
+	/**
+	 * Process the creation/insertion of a new resource object.
+	 * This should be a POST.
+	 *
+	 * @return string
+	 */
+	public function create()
+	{
+		return lang('RESTful.notImplemented', ['create']);
+	}
+
+	/**
+	 * Present a view to confirm the deletion of a specific resource object
+	 *
+	 * @param  type $id
+	 * @return string
+	 */
+	public function remove($id = null)
+	{
+		return lang('RESTful.notImplemented', ['remove']);
+	}
+
+	/**
+	 * Process the deletion of a specific resource object
+	 *
+	 * @param  type $id
+	 * @return string
+	 */
+	public function delete($id = null)
+	{
+		return lang('RESTful.notImplemented', ['delete']);
+	}
+
+	/**
+	 * Present a view to edit the properties of a specific resource object
+	 *
+	 * @param  type $id
+	 * @return string
+	 */
+	public function edit($id = null)
+	{
+		return lang('RESTful.notImplemented', ['edit']);
+	}
+
+	/**
+	 * Process the updating, full or partial, of a specific resource object.
+	 * This should be a POST.
+	 *
+	 * @param  type $id
+	 * @return string
+	 */
+	public function update($id = null)
+	{
+		return lang('RESTful.notImplemented', ['update']);
+	}
+
+	//--------------------------------------------------------------------
+
+	/**
+	 * Set or change the model this controller is bound to.
+	 * Given either the name or the object, determine the other.
+	 *
+	 * @param string|object $which
+	 */
+	public function setModel($which = null)
+	{
+		// save what we have been given
+		if (! empty($which))
+		{
+			if (is_object($which))
+			{
+				$this->model = $which;
+			}
+			else
+			{
+				$this->modelName = $which;
+			}
+		}
```

Comment:
Should'nt we force the name and object to be cleared here once something has been introduced? 
It's just that if we are assigning the model for the first time, everything is fine, but if we are changing the model it will not work well anymore because we will have the class name of the previous model and the new object or the previous object and with the name of the new class.
Maybe something like this:
if (is_object($which))
{
  $this->model = $which;
  $this->modelName = null;
}
else
{
  $this->model = null;
  $this->modelName = $which;
}